handler.py

```diff
@@ -0,0 +1,61 @@
+import logging
+import os
+from pathlib import Path
+
+import boto3
+import redis
+
+from s3_log_shipper.parsers import ParserManager
+from s3_log_shipper.s3_event_models import S3Event
+from s3_log_shipper.shipper import RedisLogShipper
+
+log: logging.Logger = logging.getLogger(__name__)
+
+
+def get_log_parser() -> ParserManager:
+    return ParserManager(config_file=get_config_file())
+
+
+def get_config_file() -> Path:
+    default = f"{os.path.dirname(__file__)}/input_files.json"
+    path = os.environ.get("CONFIG_FILE", default)
+    file = Path(path)
+    if not file.exists() or not file.is_file():
+        msg = f"Expected to find parser config file at {file.absolute()}. No such file found."
+        log.error(msg)
+        raise Exception(msg)
+    return file
+
+
+def get_output_redis_host_from_environment():
+    try:
+        return os.environ["REDIS_HOST"]
+    except KeyError as e:
+        raise Exception("env variable is not found: {}".format(e))
+
+
+def get_output_redis_port_from_environment():
+    try:
+        return os.environ["REDIS_PORT"]
+    except KeyError as e:
+        raise Exception("env variable is not found: {}".format(e))
+
+
+S3_CLIENT = boto3.client("s3")
```

Comment:
I'm not 100% certain about whether we can create a client in the global space without its access creds expiring - think those are generated automatically when the client is created, but they be time-limited - worth a double-check.

Reply:
> When you specify a profile that has IAM role configuration, boto3 will make an AssumeRole call to retrieve temporary credentials. Subsequent boto3 API calls will use the cached temporary credentials until they expire, in which case boto3 will automatically refresh credentials. boto3 does not write these temporary credentials to disk. This means that temporary credentials from the AssumeRole calls are only cached in memory within a single Session. All clients created from that session will share the same temporary credentials.

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#assume-role-provider

Reply:
In other words, it'll refresh the credentials automatically whilst the object is kept warm in memory, and from a cold start it doesn't matter anyway